### PR TITLE
feat: Resume session replay when app enters foreground

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

The session replay pauses when the app enters background, now it resumes when back to the foreground.

## :green_heart: How did you test it?

UITest and samples

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: Sources/Sentry/SentrySessionReplayIntegration.m [hunk 1]
```diff
@@ -141,6 +139,11 @@ - (void)startWithOptions:(SentryReplayOptions *)replayOptions
                                                    name:UIApplicationDidEnterBackgroundNotification
                                                  object:nil];
 
+        [NSNotificationCenter.defaultCenter addObserver:self
```
Comment (philipphofmann) [inline Sources/Sentry/SentrySessionReplayIntegration.m:142]: `m`: Again, I'm not a fan of using the NSNotificationCenter directly as it cause multiple problems with testing in the past. I would prefer to use `SentryNSNotificationCenterWrapper`.

**Review by philipphofmann — APPROVED** [review]
LGTM

File: Sources/Sentry/SentrySessionReplayIntegration.m (hunk 1 above)
Comment (brustolin) [reply Sources/Sentry/SentrySessionReplayIntegration.m:142]: Technically, there is no reason for NSNotificationCenter to cause any problems. 
Maybe the old flakiness was due to issues in the test. 
I will monitor this to ensure it does not become flaky.

Comment (philipphofmann) [reply Sources/Sentry/SentrySessionReplayIntegration.m:142]: No clue. It's just what I experienced in the past.

**Pull request merged**